src/fsharp/NameResolution.fs

```diff
@@ -696,13 +696,22 @@ let rec AddModuleOrNamespaceRefsToNameEnv g amap m root ad nenv (modrefs: Module
 and AddModuleOrNamespaceContentsToNameEnv (g:TcGlobals) amap (ad:AccessorDomain) m root nenv (modref:ModuleOrNamespaceRef) = 
      let pri = NextExtensionMethodPriority()
      let mty = modref.ModuleOrNamespaceType
```

Comment:
this bellow need careful review, can't easily assert if it does what it was doing before, but looks better!